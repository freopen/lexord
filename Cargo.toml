[package]
name = "lexord"
version = "0.1.0"
edition = "2021"

[workspace]
members = ["lexord_derive", "fuzz", "golden"]

[dependencies]
thiserror = "1.0.40"
lexord_derive = { version = "0.1.0", path = "lexord_derive" }

[dev-dependencies]
afl = "0.14.3"
arbitrary = { version = "1.3.1", features = ["derive"] }
insta = "1.31.0"
