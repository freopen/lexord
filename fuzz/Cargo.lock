# This file is automatically @generated by Cargo.
# It is not intended for manual editing.
version = 3

[[package]]
name = "arbitrary"
version = "1.3.0"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "e2d098ff73c1ca148721f37baad5ea6a465a13f9573aba8641fbbbae8164a54e"

[[package]]
name = "cc"
version = "1.0.79"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "50d30906286121d95be3d479533b458f87493b30a4b5f79a607db8f5d11aa91f"
dependencies = [
 "jobserver",
]

[[package]]
name = "jobserver"
version = "0.1.26"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "936cfd212a0155903bcbc060e316fb6cc7cbf2e1907329391ebadc1fe0ce77c2"
dependencies = [
 "libc",
]

[[package]]
name = "lexord"
version = "0.1.0"
dependencies = [
 "lexord_derive",
 "thiserror",
]

[[package]]
name = "lexord-fuzz"
version = "0.0.0"
dependencies = [
 "lexord",
 "libfuzzer-sys",
]

[[package]]
name = "lexord_derive"
version = "0.1.0"
dependencies = [
 "proc-macro2",
 "quote",
 "syn",
]

[[package]]
name = "libc"
version = "0.2.147"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "b4668fb0ea861c1df094127ac5f1da3409a82116a4ba74fca2e58ef927159bb3"

[[package]]
name = "libfuzzer-sys"
version = "0.4.6"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "beb09950ae85a0a94b27676cccf37da5ff13f27076aa1adbc6545dd0d0e1bd4e"
dependencies = [
 "arbitrary",
 "cc",
 "once_cell",
]

[[package]]
name = "once_cell"
version = "1.18.0"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "dd8b5dd2ae5ed71462c540258bedcb51965123ad7e7ccf4b9a8cafaa4a63576d"

[[package]]
name = "proc-macro2"
version = "1.0.66"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "18fb31db3f9bddb2ea821cde30a9f70117e3f119938b5ee630b7403aa6e2ead9"
dependencies = [
 "unicode-ident",
]

[[package]]
name = "quote"
version = "1.0.32"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "50f3b39ccfb720540debaa0164757101c08ecb8d326b15358ce76a62c7e85965"
dependencies = [
 "proc-macro2",
]

[[package]]
name = "syn"
version = "2.0.27"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "b60f673f44a8255b9c8c657daf66a596d435f2da81a555b06dc644d080ba45e0"
dependencies = [
 "proc-macro2",
 "quote",
 "unicode-ident",
]

[[package]]
name = "thiserror"
version = "1.0.44"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "611040a08a0439f8248d1990b111c95baa9c704c805fa1f62104b39655fd7f90"
dependencies = [
 "thiserror-impl",
]

[[package]]
name = "thiserror-impl"
version = "1.0.44"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "090198534930841fab3a5d1bb637cde49e339654e606195f8d9c76eeb081dc96"
dependencies = [
 "proc-macro2",
 "quote",
 "syn",
]

[[package]]
name = "unicode-ident"
version = "1.0.11"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "301abaae475aa91687eb82514b328ab47a211a533026cb25fc3e519b86adfc3c"
