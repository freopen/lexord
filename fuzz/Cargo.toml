[package]
name = "lexord-fuzz"
version = "0.1.0"
edition = "2021"

[dependencies]
lexord = { version = "0.1.0", path = ".." }
afl = "0.14.3"
arbitrary = { version = "1.3.1", features = ["derive"] }
