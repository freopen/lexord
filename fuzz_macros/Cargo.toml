[package]
name = "lexord_fuzz_macros"
version = "0.1.0"
edition = "2021"
publish = false

[lib]
proc-macro = true

[dependencies]
itertools = "0.13.0"
proc-macro2 = "1.0.85"
quote = "1.0.36"
syn = { version = "2.0.66", features = ["extra-traits", "fold"] }
