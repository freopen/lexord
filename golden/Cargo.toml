[package]
name = "golden"
version = "0.1.0"
edition = "2021"

[lib]
proc-macro = true

[dependencies]
arbitrary = "1.3.2"
lexord = { version = "0.1.0", path = ".." }
lexord-fuzz = { version = "0.1.0", path = "../fuzz" }
proc-macro2 = "1.0.65"
quote = "1.0.30"
syn = { version = "2.0.26", features = ["extra-traits"] }
